client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt

```diff
@@ -1,10 +1,77 @@
 package org.thechance.common.presentation.restaurant
 
-import cafe.adriel.voyager.core.model.StateScreenModel
-import org.thechance.common.presentation.uistate.RestaurantUiState
+import org.thechance.common.domain.entity.Restaurant
+import org.thechance.common.domain.usecase.IGetRestaurantsUseCase
+import org.thechance.common.presentation.base.BaseScreenModel
+import org.thechance.common.presentation.util.ErrorState
+import kotlin.math.ceil
 
 
-class RestaurantScreenModel : StateScreenModel<RestaurantUiState>(RestaurantUiState()) {
+class RestaurantScreenModel(
+    private val getRestaurants: IGetRestaurantsUseCase
+) : BaseScreenModel<RestaurantUiState, RestaurantUIEffect>(RestaurantUiState()),
+    RestaurantInteractionListener {
 
+    init {
+        getRestaurant()
+    }
 
-}
\ No newline at end of file
+    private fun getRestaurant() {
+        tryToExecute(getRestaurants::getRestaurants, ::onGetRestaurantSuccessfully, ::onError)
+    }
```

Comment:
```suggestion
    private fun getRestaurants() {
        tryToExecute(getRestaurants::getRestaurants, ::onGetRestaurantSuccessfully, ::onError)
    }
```
the function name is reffereing to get a single restaurant